src/metabase/native_query_analyzer.clj

```diff
@@ -0,0 +1,83 @@
+(ns metabase.native-query-analyzer
+  "Integration with Macaw, which parses native SQL queries. All SQL-specific logic is in Macaw, the purpose of this
+  namespace is to:
+
+  1. Translate Metabase-isms into generic SQL that Macaw can understand.
+  2. Contain Metabase-specific business logic."
+  (:require
+   [clojure.string :as str]
+   [macaw.core :as mac]
+   [metabase.config :as config]
+   [metabase.util :as u]
+   [metabase.util.log :as log]
+   [toucan2.core :as t2]))
+
+(def ^:dynamic *parse-queries-in-test?*
+  "Normally, a native card's query is parsed on every create/update. For most tests, this is an unnecessary
+  expense. Therefore, we skip parsing while testing unless this variable is turned on. c.f. [[active?]]"
+  false)
+
+(defn- active?
+  "Should the query run? Either we're not testing or it's been explicitly turned on.
+
+  c.f. [[*parse-queries-in-test?*]]"
+  []
+  (or (not config/is-test?)
+      *parse-queries-in-test?*))
+
+(defn- normalize-name
+  ;; TODO: This is wildly naive and will be revisited once the rest of the plumbing is sorted out
+  ;; c.f. Milestone 3 of the epic: https://github.com/metabase/metabase/issues/36911
+  [name]
+  (-> name
+      (str/replace "\"" "")
+      u/lower-case-en))
+
+(defn- fields-for-query
+  "Very naively selects Fields that could be used in the query. Improvements to this are planned for Q2 2024,
+  c.f. Milestone 3 of https://github.com/metabase/metabase/issues/36911"
+  [q db-id]
+  (try
+    (let [{:keys [columns tables]} (mac/query->components (mac/parsed-query q))]
+      (t2/query {:select [[:f/id :field_id] [:f/name :column_name] [:t/name :table_name]]
```

Comment:
Inconsistent use of `.` and `/` in this query. FWIW I prefer `.`

Reply:
fixed. The `%lower` stuff needs to be that way, AFAIU